### DISCUSSION TITLE
Good Settings for Starling Murmurations

### DISCUSSION THREAD

**Discussion by Joel-BB** [Shadertech/UE5NiagaraComputeExample#2] [Q&A] [answered]
It is possible to get starling murmurations using this plugin? I've been playing around with the settings but no matter what they always seem to clump into balls and then stick together and never separate back out. Thanks in advance.

**Comment by AaronTrotter** [comment, marked as answer]
Hello!

For the example, I want to keep it as simple as can be so that it can be used as boilerplate. Boids are currently only comparing against each other, nothing fancy and that's why they are always clumped together. The main purpose of the repo was to show how to write for the RDG and show how to get custom buffers into Niagara and materials.

If you want to introduce patterns then that's on you to implement; its easy:
- if you are using the niagara-only example then just add a noise modifier to the Niagara system.
- if you are using the compute shader example then just include a shader noise library and apply the force before the position is updated in the UpdateBoids kernel of CS_Boids.usf. It is technically possible to just add some forces directly on the niagara system but ideally you would need to grab this data back and update the boids buffer with the new positions. Unreal has an export particle data function for that but I dont recommend doing things that way.


P.S. Make sure you have the latest changes as I have refactored a lot last weekend and fixed some small bugs.
